### ISSUE TITLE
apollo-server-lambda v2 CORS issue

### ISSUE THREAD

**Issue by pulpfree** [apollographql/apollo-server#1307]
I'm using a apollo-server-lambda (2.0.0-rc.6) and running into a problem with cors. When the browser does a preflight test with the `OPTIONS` method, the `Access-Control-Allow-Origin` header is missing. My config is as outlined in the [lambda deployment](https://www.apollographql.com/docs/apollo-server/v2/deployment/lambda.html) :
```
exports.graphqlHandler = server.createHandler({
  cors: {
    origin: '*',
    credentials: true,
  },
})
```
All is fine when testing with curl or insomnia, returning the right access header, but of course the browser does a preflight and it fails when doing the `OPTIONS` method. When I try the same thing with just the `apollo-server` it does return the `Access-Control-Allow-Origin` header on the preflight. However not so with apollo-server-lambda which has me thinking this is a bug.

I am deploying this with AWS SAM and I've tried enabling cors in the API Gateway console, but I believe this needs to happen at the server level.

**Comment by adnsio** [comment]
@evans @pulpfree We could handle OPTIONS requests directly from the `apollo-server-lambda` to reply with the same CORS that will be used by GET and POST. Tell me what you think 😄

**Comment by pulpfree** [comment]
@adnsio That makes sense to me, though my experience/understanding with `apollo-server-lambda` is limited. I'm assuming that it would be a necessary requirement to handle the OPTIONS request in a similar manner, how else can the 'preflight' work, just saying.

**Comment by adnsio** [comment]
@evans It will be also easier to use because it will be enough to set an ANY request handler on AWS API Gateway and `apollo-server-lambda` will do everything.

**Comment by pulpfree** [comment]
@adnsio @evans Certainly makes sense to me, a simpler SAM template is good!

**Comment by evans** [comment]
@adnsio That makes total sense! Definitely like the simplicity. Are you able to open a PR?

**Comment by adnsio** [comment]
@evans Sure, I will work within this week.

**Comment by adnsio** [comment]
@evans @pulpfree Done in #1344

**Comment by evans** [comment]
Fixed with #1344!

**Comment by pulpfree** [comment]
@evans @adnsio I've done some testing and here's the error I'm receiving with Chrome: 
`Request header field content-type is not allowed by Access-Control-Allow-Headers in preflight response`.
 This occurs on the `OPTIONS` method
